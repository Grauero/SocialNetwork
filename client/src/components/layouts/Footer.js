import React from 'react';

const Footer = () => (
  <footer className="bg-light text-dark mt-5 p-4 text-center">
    Copyright &copy; {new Date().getFullYear()} Network
  </footer>
);

export default Footer;
