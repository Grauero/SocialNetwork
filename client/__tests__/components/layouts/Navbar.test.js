import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { Navbar } from '../../../src/components/layouts/Navbar';

const props = {
  auth: { isAuthenticated: true, user: {} },
  logoutUser: jest.fn(),
  clearCurrentProfile: jest.fn()
};
let component;

beforeEach(
  () =>
    (component = mount(
      <BrowserRouter>
        <Navbar {...props} />
      </BrowserRouter>
    ))
);

afterEach(() => component.unmount());

it('calls clearCurrentProfile and logoutUser by pressing logout button', () => {
  const logoutButton = component.find('a').get(4);
  logoutButton.props.onClick({ preventDefault: jest.fn() });

  expect(props.clearCurrentProfile).toHaveBeenCalled();
  expect(props.clearCurrentProfile).toHaveBeenCalledTimes(1);
  expect(props.logoutUser).toHaveBeenCalled();
  expect(props.logoutUser).toHaveBeenCalledTimes(1);
});

it('should render auth links if user is authenticated', () => {
  const authLinks = component.find('Link');

  expect(toJSON(authLinks)).toMatchSnapshot();
});

it('should render guest links if user is NOT authenticated', () => {
  const guestLinks = component.find('Link');

  expect(toJSON(guestLinks)).toMatchSnapshot();
});
