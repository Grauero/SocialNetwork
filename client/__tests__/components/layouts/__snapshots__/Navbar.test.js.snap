// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders auth links if user is authenticated 1`] = `
Array [
  <Link
    className="navbar-brand"
    replace={false}
    to="/"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      DevNetwork
    </a>
  </Link>,
  <Link
    className="nav-link"
    replace={false}
    to="/profiles"
  >
    <a
      className="nav-link"
      href="/profiles"
      onClick={[Function]}
    >
      Developers
    </a>
  </Link>,
  <Link
    className="nav-link"
    replace={false}
    to="/feed"
  >
    <a
      className="nav-link"
      href="/feed"
      onClick={[Function]}
    >
      Post Feed
    </a>
  </Link>,
  <Link
    className="nav-link"
    replace={false}
    to="/dashboard"
  >
    <a
      className="nav-link"
      href="/dashboard"
      onClick={[Function]}
    >
      Dashboard
    </a>
  </Link>,
]
`;

exports[`renders guest links if user is NOT authenticated 1`] = `
Array [
  <Link
    className="navbar-brand"
    replace={false}
    to="/"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      DevNetwork
    </a>
  </Link>,
  <Link
    className="nav-link"
    replace={false}
    to="/profiles"
  >
    <a
      className="nav-link"
      href="/profiles"
      onClick={[Function]}
    >
      Developers
    </a>
  </Link>,
  <Link
    className="nav-link"
    replace={false}
    to="/feed"
  >
    <a
      className="nav-link"
      href="/feed"
      onClick={[Function]}
    >
      Post Feed
    </a>
  </Link>,
  <Link
    className="nav-link"
    replace={false}
    to="/dashboard"
  >
    <a
      className="nav-link"
      href="/dashboard"
      onClick={[Function]}
    >
      Dashboard
    </a>
  </Link>,
]
`;
